### ISSUE TITLE
Common permissions require deprecated provider syntax

### ISSUE THREAD

**Issue by kpeters-cbsi** [AntonBazhal/serverless-plugin-custom-roles#16]
Serverless have deprecated this syntax for specifying IAM role statements in the `provider` section of `serverless.yml`:

```yaml
provider:
  name: aws
  iamRoleStatements:
    - Effect: Allow
      Action: service:Action
      Resource: arn:aws:service:us-east-1:123456789012:something/*
``` 

in favor of:

```yaml
provider:
  name: aws
  iam:
    role:
      statements:
        - Effect: Allow
          Action: service:Action
          Resource: arn:aws:service:us-east-1:123456789012:something/*
``` 

However, this plugin still requires the deprecated syntax in order to add permissions common to all functions.

**Comment by AntonBazhal** [comment]
Hello, yes I'm aware of this issue, but I've never got time to deal with it, unfortunately. PRs are welcome :)

**Comment by paulcruse-syn** [comment]
Hi,

I have fixed this issue and created a branch and I am trying to make a PR but I don't think you have allowed anyone to push a new branches, this is a setting you must change in your repo settings.

Please open up the permissions so that others may create branches, push to them, and then create pull requests.

To give a summary on what I did.

`index.js:L194`:
```javascript
let sharedRoleStatements = null;
    if (service.provider.iam && service.provider.iam.role && service.provider.iam.role.statements) {
      sharedRoleStatements = service.provider.iam.role.statements;
    } else if (service.provider.iamRoleStatements) {
      sharedRoleStatements = service.provider.iamRoleStatements;
    }
```

`test/index-test.js:L559`
```javascript
it('should add shared policy when provider has iam role statements defined', function() {
      const statements = [{
        Effect: 'Allow',
        Action: [
          'xray:PutTraceSegments',
          'xray:PutTelemetryRecords'
        ],
        Resource: '*'
      }];
      const instance = createTestInstance({
        provider: {
          iam: {
            role: {
              statements
            }
          }
        },
        functions: {
          function1: {}
        }
      });

      instance.createRoles();

      expect(instance)
        .to.have.nested.property('serverless.service.resources')
        .that.containSubset({
          Resources: {
            Function1LambdaFunctionRole: {
              Type: 'AWS::IAM::Role',
              Properties: {
                Policies: [{
                  PolicyName: 'shared',
                  PolicyDocument: {
                    Version: '2012-10-17',
                    Statement: statements
                  }
                }]
              }
            }
          }
        });

      sinon.assert.notCalled(instance.serverless.cli.log);
    });
```

It follows your linting standards as well as your test coverage standards

**Comment by paulcruse3** [comment]
Hi, 

I created PR to solve this issue:  https://github.com/AntonBazhal/serverless-plugin-custom-roles/pull/21

**Comment by AntonBazhal** [comment]
Thanks for your contribution. Released in `3.0.0`.
